src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java

```diff
@@ -167,11 +171,18 @@ private static void writeArray(JsonWriter writer, String name, Object value, Sch
         String.format("Field '%s' is of value null, which is not a valid value for BigQuery type array.", name));
     }
 
-    if (!(value instanceof Collection) && !value.getClass().isArray()) {
-      throw new IllegalArgumentException("The value should be of type collection or array. Got: " + value.getClass());
+    Collection collection;
+    if (value instanceof Collection) {
+      collection = (Collection) value;
+    } else if (value instanceof Object[]) {
+      collection = Arrays.asList((Object[]) value);
+    } else {
+      throw new IllegalArgumentException(String.format(
+        "The value is of type '%s', expected Collection or array.", value.getClass().getSimpleName()));
```

Comment:
This is an existing issue, but the message should include the name of the field.

A value for field '%s' is of type '%s' when it is expected to be a Collection or array.

Reply:
Done.